From 6286e55b62db40006cddeb98dbdb0b46695283a0 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 30 Sep 2022 11:59:52 +0300
Subject: [PATCH] [SW-69872] gaudi3: d2d: move d2d init call to end of recovery

The d2d link init call in rom recovery is moved to end of
recovery. This is in preparation to introduce a synchronization
between master and slave d2d init such that slave is done with
all the processing before waiting for d2d init.

Change-Id: I819aa29a1edc6ff53e9b871f631938d7f651da84
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index ea99e56..d46c72f 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -217,12 +217,12 @@
 	security_rr_open_debug_if_config();
 	rom_pcie_recovery();
 
-	rom_d2d_link_init();
-
 	if ((arcboot_rom_global.config_fuse &
 		EFUSE_SYSCFG_DISABLE_TWI_IF_MASK) == 0)
 		security_i2c_s_enable();
 
+	rom_d2d_link_init();
+
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_RECOVERY_END);
 }
 
